sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -432,7 +442,13 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     else {
         ng_ipv6_addr_t *next_hop = NULL;
 
-        next_hop = &hdr->dst;   /* TODO: next hop determination */
+#ifdef MODULE_NG_IPV6_EXT_RH
+        next_hop = ng_ipv6_ext_rh_next_hop(hdr);
```

Comment:
Okay I don't get that. How is it specified as a destination and by whom? Who caches the SRH beforehand so I can use it here? And would some kind of SRH cache add code overhead too?